app/src/main/AndroidManifest.xml

```diff
@@ -1,7 +1,14 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="edu.uc.kovaciad.slicetracker">
-
+    <!--Give permission-->
+    <uses-permission android:name="android.permission.BLUETOOTH" />
+    <uses-permission android:name="android.permission.BLUETOOTH_ADMIN" />
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+    <!--Give Internet permission-->
+    <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
```

Comment:
Giving Bluetooth and Internet permission, which are included in the AndroidManifest.xml. Bluetooth is necessary to perform any Bluetooth communication, such as requesting a connection, accepting a connection, and transferring data. Check out : https://developer.android.com/guide/topics/connectivity/bluetooth/permissions#features
Also, your app might need internet permission to operate, so I also added that. Check out: https://developer.android.com/training/basics/network-ops/connecting

Reply:
Is Bluetooth actually used in this application?  I don't see it anywhere.  If it is not used, then adding the permission is superfluous.  You're asking the user to give you permission for something you won't use, and a user may be reluctant to grant that permission.